### PR TITLE
Kafka chart use confluent kafka

### PR DESCRIPTION
**Pull request by benjigoldberg**
…e a true liveness check

@viglesiasce This switches from the solsson/kafka chart to confluentinc/cp-kafka. Because of this change, we can now pass most configuration to kafka as simple environment variables. 

Hightlights of this PR:
* A new test for the chart that creates a topic, produces a message, and then consumes it, verifying the cluster has come up cleanly
* Users may now add or change kafka settings via a `configurationOverrides` setting in the values file. This is placed into a `configmap` which is eventually rendered into the pods as env vars. 
* Liveness does a simple TCP check instead of a heavy-handed broker connection / topic list (which is now used only in the readiness check)
* The confluent image requires `advertised.listeners`, which with a k8s headless service is the Pod IP. This is simple enough to get via the downward API, but there's enough rendering going on with the Kafka settings that I was required to slap it into the `sh -c` command as well.
* The one last bit thats worth noting is that broker id/stateful set pod number cant currently be captured via the downward API, so I had to override the container's CMD to parse hostname for the number of the pod/broker id setting.

### CODE REVIEW

File: incubator/kafka/templates/tests/test_topic_create_consume_produce.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    - -c
+    - |
+      # Create the topic
+      kafka-topics --zookeeper {{ template "zookeeper.url" . }} --topic helm-test-topic-create-consume-produce --create --partitions 1 --replication-factor 1 --if-not-exists && \
+      # Create a message
+      MESSAGE="`date -u`" && \
+      # Produce a test message to the topic
+      echo $MESSAGE | kafka-console-producer --broker-list {{ template "kafka.fullname" . }}:9092 --topic helm-test-topic-create-consume-produce && \
+      # Produce a test message to the topic
+      kafka-console-producer --broker-list {{ template "kafka.fullname" . }}-headless:9092 --topic helm-test-topic-create-consume-produce && \
```
Comment (benjigoldberg) [inline incubator/kafka/templates/tests/test_topic_create_consume_produce.yaml]: Rm lines 21-22

Comment (viglesiasce) [reply incubator/kafka/templates/tests/test_topic_create_consume_produce.yaml]: @benjigoldberg are you wanting to do this before this gets merged?

Comment (benjigoldberg) [reply incubator/kafka/templates/tests/test_topic_create_consume_produce.yaml]: @viglesiasce yeah. I pushed this up and then noticed this on my commute. I was tinkering with the tests and forget to delete these two lines

**Comment by viglesiasce** [comment]
Thanks again @benjigoldberg!!!

**Pull request merged**